crates/sequencer/src/sequencer.rs

```diff
@@ -630,4 +663,79 @@ where
         rpc_methods.merge(rpc).unwrap();
         Ok(rpc_methods)
     }
+
+    pub async fn insert_commitment_info_to_db(
+        &self,
+        tx_id: OneshotReceiver<Result<<Da as DaService>::TransactionId, <Da as DaService>::Error>>,
+        db_config: OffchainDbConfig,
+        l1_start_height: u64,
+        l1_end_height: u64,
+        commitment: SequencerCommitment,
+        l2_range: RangeInclusive<BatchNumber>,
+    ) {
+        // spawn an async task in the background and await the tx_id then save to pg
+        tokio::spawn(async move {
+            match tx_id.await {
+                Ok(Ok(tx_id)) => match PostgresConnector::new(db_config).await {
+                    Ok(pg_connector) => {
+                        info!("Sequencer: commitment tx_id: {:?}", tx_id.to_string());
+                        pg_connector
+                            .insert_sequencer_commitment(
+                                l1_start_height as u32,
+                                l1_end_height as u32,
+                                tx_id.to_string(),
+                                commitment.l1_start_block_hash.to_vec(),
+                                commitment.l1_end_block_hash.to_vec(),
+                                l2_range.start().0 as u32,
+                                (l2_range.end().0 + 1) as u32,
+                                commitment.merkle_root.to_vec(),
+                                "mempool".to_string(),
+                            )
+                            .await
+                            .expect("Sequencer: Failed to insert sequencer commitment");
+                    }
+                    Err(e) => {
+                        warn!("Failed to connect to postgres: {:?}", e);
+                    }
+                },
+                _ => {
+                    warn!("Sequencer: Failed to submit commitment: ");
+                }
```

Comment:
if we are here we should resubmit the commitmnet, because we already modified our ledger db according to our commitment

Reply:
maybe if we set ledger_db here, then if we can't commit, we don't save to ledger db and we try next time?

idk, let's discuss